main.tf

```diff
@@ -443,7 +443,7 @@ resource "aws_subnet" "elasticache" {
 resource "aws_elasticache_subnet_group" "elasticache" {
   count = var.create_vpc && length(var.elasticache_subnets) > 0 && var.create_elasticache_subnet_group ? 1 : 0
 
-  name        = var.name
+  name        = var.elasticache_subnet_group_name != null ? lower(var.elasticache_subnet_group_name) : lower(var.name)
```

Comment:
You added `lower(var.name)`. It may break infrastructure of the current users of this module during upgrade t the new version.
Not sure you need add `lower()` here at all.